applications/ShapeOptimizationApplication/python_scripts/custom_ios/wrl_reader.py

```diff
@@ -0,0 +1,95 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Geiser Armin, https://github.com/armingeiser
+#
+# ==============================================================================
+
+class Shape:
+    def __init__(self, name, nodes, triangles):
+        self.name = name
+        self.nodes = nodes
+        self.triangles = triangles
+
+
+def read_nodes(line, file):
+    nodes = []
+    while not "[" in line:
+        line = next(file)
+
+    line = next(file)
+
+    while not "]" in line:
+        if line == "\n":
+            line = next(file)
+
+        try:
+            entries = line.split(",")[0].split()
+            if len(entries) != 3:
+                raise RuntimeError("wrl_reader: Did not find 3 coordinate components!", line)
+            nodes.append([float(x) for x in entries])
+        except IOError:
+            pass
+
+        line = next(file)
+
+    return nodes
+
+
+def read_triangles(line, file):
+    triangles = []
+    while not "[" in line:
+        line = next(file)
+
+    line = next(file)
+
+    while not "]" in line:
+        if line == "\n":
+            line = next(file)
+
+        try:
+            entries = line.split(",")[:3]
+            if len(entries) != 3:
+                raise RuntimeError("wrl_reader: Did not find 3 triangle node indices!", line)
+            triangles.append([int(x) for x in entries])
+        except IOError:
+            pass
+
+        line = next(file)
+
+    return triangles
+
+
+def read_shape(line, file):
+    name = line.split("#")[1].strip()
+    nodes = []
+    triangles = []
+    while not nodes or not triangles:
+        if line.strip().startswith("coord Coordinate"):
+            nodes = read_nodes(line, file)
+        elif line.strip().startswith("coordIndex"):
+            triangles = read_triangles(line, file)
+        line = next(file)
+
+    shape = Shape(name, nodes, triangles)
```

Comment:
Here you could directly return the created shape to save a few lines or letters. This requires to remove the other return two lines below.

```suggestion
    return Shape(name, nodes, triangles)
```